server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -287,6 +287,19 @@ static void parseObjectOrNested(DocumentParserContext context) throws IOExceptio
         }
 
         if (context.parent().isNested()) {
+            if (context.mappingLookup().isSourceSynthetic() && context.getClonedSource() == false) {
+                Tuple<DocumentParserContext, XContentBuilder> tuple = XContentDataHelper.cloneSubContext(context);
+                context.addIgnoredField(
+                    new IgnoredSourceFieldMapper.NameValue(
+                        context.parent().name(),
+                        context.parent().fullPath().indexOf(context.parent().simpleName()),
+                        XContentDataHelper.encodeXContentBuilder(tuple.v2())
+                    )
+                );
+                context = tuple.v1();
+                token = context.parser().currentToken();
+                parser = context.parser();
+            }
             context = context.createNestedContext((NestedObjectMapper) context.parent());
```

Comment:
Should the `DocumentParserContext` constructor that is used eventually from `createNestedContext(...)` method also copy `clonedSource` field from provided context?

Reply:
Good catch, this needs to be propagated.